### ISSUE TITLE
Nice project, would be nicer if it didn't exist. 

### ISSUE THREAD

**Issue by UmbraAtrox** [CryptoCTB/LiberatingMarsCLI#2]
I doubt its good to support chitu sales by making community fixes, you only make people shove more money into a anti-opensource company, since they never feel any consequence when they always can turn to the community for a patch. 

If the question is: Why can't i use another slicer with my printer?
The answer should be: Because the company is you bought it from is shady af and you should think twice in the future if you ever want to buy from them again. 

If software like LiberatingMars workarounds the problem before the question gets asked there isn't a reason to not buy yet another chitu printer and also no reason to support opensource machines. Imagine if instead of reprap FDM went this way. We would have had nothing but proprietary blackboxes locked behind paywalls and subscriptions and as fixable as your typical desktop inkjet printer. 

So while i am impressed by your project i still have to ask you to please stop for the sake of anyone who wants 3dprinting stay open to everyone.

**Comment by marknn3** [comment]
I totally agree with you, and (partial) opensourceing makes the community more involved and can have many benefits.
But for people who already have this printer this project can be very useful.
Did you know the free LycheeSlicer now has official support for Elegoo Mars 3? (and most other Elegoo models!)
In that respect Elegoo is opening up and making their proprietary file format accessible for other slicers.

**Comment by tslater2006** [comment]
Nitpick here, it wasn't Elegoo that opened up but rather Lychee caved to
licensing the Chitubox SDK to be able to produce compatible CTB files.
Elegoo really just got caught in the storm by being the first to use the
new encrypted format. Wasn't their fault nor something they could control.

**Comment by UmbraAtrox** [comment]
> I totally agree with you, and (partial) opensourceing makes the community more involved and can have many benefits.
> But for people who already have this printer this project can be very useful.

Sure, but only in short term, in the exact moment they are stuck with that printer, but in the long run in only benefits chitu since  you only remove the downsites from chitus predatory sheme without chitu actually changing anything. I understand why you think it benefits the people, but in reality it really doesn't. If we want this bs chitu does to stop they have to die to their own mistakes, which won't happen if the community fixes them before they can do any damage. And by damage i mean sales numbers which is the only language they speak.

Its the same issue as with jailbreaking iphones. The solution isn't to jailbreak but to not buy into the jail. Who did it really benefit? Since you could just remove the downsides people still bought into their ecosystem, still bought their hardware, still spent on the appstore and in the long run it did nothing but benefit apple. Not talking about those who would have bought it anyway, but the small percentage who were concidering or would have concidered something else if they couldn't just yellowsnow the issues away. 

Before this project existed there was a reason to not buy eg. a Mars 3, because it would have been locked to chitubox, now there isn't. So people have no reason to even consider anything else and therefore throw more money at elegoo and ultimately chitu. 

> Elegoo really just got caught in the storm by being the first to use the new encrypted format. Wasn't their fault nor something they could control.

Elegoo abosulutely had control by choosing to not invest what they earned into an open platform instead sticking to chitu even though they went the proprietary file format, don't provide schematics, datasheets or source. everything that has been aparent since the first mars released 3 years ago. 
I totally get buying a non optimal 3rd party solution for the start but continuing to use chitu, choosing to not spend a penny on research but pocketing it all is definetly something they have control over. Which in intself is incredibly shortsighted from elegoo.

They already established enough they know people buy from them so there is little risk throwing money into development
They are already able to manufacture the hardware. 
All thats missing is pcb and firmware. If they cant do it themselves just throw out a bounty for the team that solves it. All the pcb has to control is that lcd, one stepper, one endstop where the lcd is the hard part. As soon as the first iteration is out there will be developers all over the globe willing to contribute.

They would be the SLAcommunities' favorite brand over night, save costs by not paying chitu, and even make aditional money by selling that new open pcb as an upgrade for their old chitu based printers. 

And while i believe the community can develop their own free pcb and firmware for the old chitu machines it would be just another patch, you can not run around an put out fires where they happen for the sake of helping. You have to make them build in a way fires don't happen, if they don't care you have to let it burn down. Damage control always has to happen after the cause has been fixed else you only spend your time controlling the damage while improving nothing.

**Comment by PaulGD03** [comment]
LiberatingMarsCLI has been released due to two major reasons. This was primarily my decision, and I will explain it.<br>
1. ChiTu Systems decided to release a “SDK” with a “license agreement” - something that I think everyone can agree is really just an evil contract against open source - that you could sign up for as an individual. At this point, they tried convincing people that their method solves the “lock-in issue”. As I do not agree with their measures, I provided this as an open source alternative to decrease the amount of people intending to sign up for the SDK.<br>
2. Mango released LycheeSlicer 3.6 with the functionality of the CTBv4E format using the SDK provided by ChiTu. With their main competitor basically “caving in”, the “fight” against the locking-in is basically lost, ChiTu pushed through their will and succeeded.<br><br>
So at this point, LiberatingMars(CLI) is basically obsolete, I won’t finish the missing features / errors anymore. This project will stay open-source for educational purpose and to document the format, but it will be frozen in an archived form.
